mllib/src/main/scala/org/apache/spark/mllib/stat/test/ChiSqTest.scala

```diff
@@ -151,6 +155,8 @@ private[spark] object ChiSqTest extends Logging {
    */
   def chiSquared(observed: Vector,
       expected: Vector = Vectors.dense(Array.empty[Double]),
+      simulatePValue: Boolean = false,
```

Comment:
I see. This looks like a good opportunity to bring ML into alignment with MLlib since the ML implementation of Chi squared tests so far seems to only include the independence test. 
https://github.com/apache/spark/blob/master/mllib/src/main/scala/org/apache/spark/ml/stat/ChiSquareTest.scala

It will probably take me a week or so, but I could work on migrating the goodness of fit test and this p-value simulation into ML if that's the way to go. Thoughts?

Reply:
Spark depends on https://github.com/addthis/stream-lib/blob/master/src/main/java/com/clearspring/analytics/stream/quantile/TDigest.java already, and this version is also implemented by Ted. Can you check if those two versions work the same way? If so, we don't need to add another dependency which is hard to justify normally.